umibukela/results.py

```diff
@@ -0,0 +1,191 @@
+            question_results = count_options(
+                submissions,
+                child['children'],
+                deeper_path,
+                question_results,
+                None)
+        elif (child_is_type(child, 'select one')
+              or child_is_type(child, 'select all that apply')):
+            # multiple choice questions
+            current_option_counts = count_question_options(
+                submissions,
+                deeper_path
+            )
+            question_results = deep_dict_set(
+                question_results,
+                child['label'],
+                [pathstr(deeper_path), 'label']
+            )
+            question_results = count_options(
+                submissions,
+                child['children'],
+                deeper_path,
+                question_results,
+                current_option_counts
+            )
+        elif ('type' not in child):
+            # option in multiple choice question
+            question_results = deep_dict_set(
+                question_results,
+                child['label'],
+                [pathstr(path), 'options', child['name'], 'label']
+            )
+            question_results = deep_dict_set(
+                question_results,
+                idx,
+                [pathstr(path), 'options', child['name'], 'idx']
+            )
+            question_results = set_option_counts(
+                path,
+                child['name'],
+                question_results,
+                current_option_counts
+            )
+        else:
+            pass
+        idx += 1
+    return question_results
+
+
+def child_is_type(child, type):
+    return ('type' in child) and child['type'] == type
+
+
+def count_question_options(site_submissions, path):
+    cols = ['facility', 'demographics_group/gender', pathstr(path)]
+    question_table = site_submissions.loc[:, cols]
+    question_counts = question_table.groupby(
+        ['demographics_group/gender', pathstr(path)]
+    ).count()
+    return question_counts
+
+
+def set_option_counts(path, option_name, results, current_option_counts):
+    for gender in ['male', 'female']:
+        option_table = current_option_counts
+
+        try:
+            val = int(option_table.loc[gender, option_name])
+        except KeyError:
+            # values that aren't counted because they don't occur in the
+            # results for this question won't be indexes in the counts
+            print("########################################")
+            print(path, gender, option_name_as_idx)
+            print(option_table)
+            val = 0
+
+        results = deep_dict_set(
+            results,
+            val,
+            [pathstr(path), 'options', option_name, 'count', gender]
+        )
+    return results
+
+
+def pathstr(path):
+    """ / separated path from array of strings"""
+    return '/'.join(path)
+
+
+def deep_dict_set(deep_dict, value, layers):
```

Comment:
Perhaps change the ordering of the args to match python's "key, value" syntax for dictionary sets. I'd also change the name layers to key, which is an array of string-based keys.